src/middlewares/validation.middleware.ts

```diff
@@ -0,0 +1,123 @@
+import { Request, Response, NextFunction } from 'express';
+
+import { ProblemException } from '../exceptions/problem.exception';
+import { createAjvInstance } from '../utils/ajv';
+import { getAppOpenAPI } from '../utils/app-openapi';
+import { parse, prepareParserConfig, tryConvertToProblemException } from '../utils/parser';
+
+import type { ValidateFunction } from 'ajv';
+import type { AsyncAPIDocument } from '../interfaces';
+
+export interface ValidationMiddlewareOptions {
+  path: string;
+  method: 'all' | 'get' | 'post' | 'put' | 'delete' | 'patch' | 'options' | 'head',
+  documents?: Record<string, 'single' | 'list'>;
```

Comment:
Is this needed? Can't we just base the validation on the number of documents passed?

Reply:
Do you mean list/single? 🤔  I don't see any problems with this approach. Ok, I'll change it, thanks!